### PR TITLE
signup and register form validation, modal

### PR DESCRIPTION
**Pull request by rebekahkwon**

### CODE REVIEW

File: client/scripts/components/modules/Login.jsx [hunk 1]
```diff
@@ -11,60 +22,100 @@ var Login = React.createClass({
...
+    password: React.PropTypes.string
+  },
 
   getInitialState: function() {
     return {
       email: '',
       password: '',
       failedLogins: 0,
-      spinner: false
+      spinner: false,
```
Comment (brianpchsu) [inline client/scripts/components/modules/Login.jsx]: Require spinner?

**Comment by rebekahkwon** [comment]
noted.

File: client/scripts/components/modules/Signup.jsx [hunk 2]
```diff
@@ -2,62 +2,140 @@
...
+    last_name: React.PropTypes.string,
+    email: React.PropTypes.string,
+    password: React.PropTypes.string
+  },
 
   getInitialState: function() {
     return {
       first_name: '',
       last_name: '',
       email: '',
```
Comment (brianpchsu) [inline client/scripts/components/modules/Signup.jsx]: Doesn't Signup need spinner as well?

**Comment by brianpchsu** [comment]
I think all is good. After Rebekah update the spinner part, I will merge it.

**Comment by rebekahkwon** [comment]
spinner omitted from modal views.

**Comment by bportnoy** [comment]
Taking a look...

**Comment by rebekahkwon** [comment]
after merge, please update login request.body for authentication

**Pull request merged**